src/lib/UtapiClient.js

```diff
@@ -227,26 +240,26 @@ export default class UtapiClient {
         const props = [];
         const { byteLength, newByteLength, oldByteLength, numberOfObjects } =
             params;
-        const metricData = {
-            byteLength,
-            newByteLength,
-            oldByteLength,
-            numberOfObjects,
-        };
-        // Only push metric levels defined in the config, otherwise push any
-        // levels that are passed in the object
-        if (params.bucket && this.metrics.indexOf('buckets') >= 0) {
-            props.push(Object.assign({
-                bucket: params.bucket,
-                level: 'buckets',
-            }, metricData));
-        }
-        if (params.accountId && this.metrics.indexOf('accounts') >= 0) {
-            props.push(Object.assign({
-                accountId: params.accountId,
-                level: 'accounts',
-            }, metricData));
-        }
+        // We add a `service` property to any non-service level to be able to
+        // build the appropriate schema key.
+        this.metrics.forEach(level => {
```

Comment:
`forEach` was used since there could be a case in which a metric level is not included in the `params`, but is included in `this.metrics`. Such a case would result in an array returned with an `undefined` element.